--- gcc-12.1.0.bak/config.guess	2022-05-06 09:30:56.000000000 +0200
+++ gcc-12.1.0/config.guess	2022-05-08 19:31:19.836425784 +0200
@@ -1109,16 +1109,24 @@
 	exit ;;
     x86_64:Linux:*:*)
 	set_cc_for_build
+	CPU=$UNAME_MACHINE
 	LIBCABI=$LIBC
 	if test "$CC_FOR_BUILD" != no_compiler_found; then
-	    if (echo '#ifdef __ILP32__'; echo IS_X32; echo '#endif') | \
+	    if (echo '#ifdef __i386__'; echo IS_X86; echo '#endif') | \
 		(CCOPTS="" $CC_FOR_BUILD -E - 2>/dev/null) | \
-		grep IS_X32 >/dev/null
+		grep IS_X86 >/dev/null
 	    then
-		LIBCABI="$LIBC"x32
+		CPU=i686
+	    else
+		if (echo '#ifdef __ILP32__'; echo IS_X32; echo '#endif') | \
+		    (CCOPTS="" $CC_FOR_BUILD -E - 2>/dev/null) | \
+		    grep IS_X32 >/dev/null
+		then
+		    LIBCABI=${LIBC}x32
+		fi
 	    fi
 	fi
-	echo "$UNAME_MACHINE"-pc-linux-"$LIBCABI"
+	echo "$CPU"-pc-linux-"$LIBCABI"
 	exit ;;
     xtensa*:Linux:*:*)
 	echo "$UNAME_MACHINE"-unknown-linux-"$LIBC"
