diff -r -u gcc-4.0.2/gcc/toplev.h gcc-4.0.2/gcc/toplev.h
--- gcc-4.0.2/gcc/toplev.h	2005-09-09 02:51:46.000000000 +0200
+++ gcc-4.0.2/gcc/toplev.h	2017-02-10 21:54:30.462957779 +0100
@@ -180,17 +180,17 @@
 #  define CTZ_HWI __builtin_ctz
 # endif
 
-extern inline int
-floor_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x ? HOST_BITS_PER_WIDE_INT - 1 - (int) CLZ_HWI (x) : -1;
-}
-
-extern inline int
-exact_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x == (x & -x) && x ? (int) CTZ_HWI (x) : -1;
-}
+extern int
+floor_log2 (unsigned HOST_WIDE_INT x);
+
+
+
+
+extern int
+exact_log2 (unsigned HOST_WIDE_INT x);
+
+
+
 #endif /* GCC_VERSION >= 3004 */
 
 /* Functions used to get and set GCC's notion of in what directory
