--- gcc-9.5.0/configure.bak	2022-05-27 09:21:10.443377037 +0200
+++ gcc-9.5.0/configure	2024-01-12 21:53:15.172764287 +0100
@@ -7561,23 +7561,23 @@
 
 # Special user-friendly check for native x86_64-linux build, if
 # multilib is not explicitly enabled.
-case "$target:$have_compiler:$host:$target:$enable_multilib" in
-  x86_64-*linux*:yes:$build:$build:)
-    # Make sure we have a development environment that handles 32-bit
-    dev64=no
-    echo "int main () { return 0; }" > conftest.c
-    ${CC} -m32 -o conftest ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} conftest.c
-    if test $? = 0 ; then
-      if test -s conftest || test -s conftest.exe ; then
-	dev64=yes
-      fi
-    fi
-    rm -f conftest*
-    if test x${dev64} != xyes ; then
-      as_fn_error $? "I suspect your system does not have 32-bit development libraries (libc and headers). If you have them, rerun configure with --enable-multilib. If you do not have them, and want to build a 64-bit-only compiler, rerun configure with --disable-multilib." "$LINENO" 5
-    fi
-    ;;
-esac
+#case "$target:$have_compiler:$host:$target:$enable_multilib" in
+#  x86_64-*linux*:yes:$build:$build:)
+#    # Make sure we have a development environment that handles 32-bit
+#    dev64=no
+#    echo "int main () { return 0; }" > conftest.c
+#    ${CC} -m32 -o conftest ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} conftest.c
+#    if test $? = 0 ; then
+#      if test -s conftest || test -s conftest.exe ; then
+#	dev64=yes
+#      fi
+#    fi
+#    rm -f conftest*
+#    if test x${dev64} != xyes ; then
+#      as_fn_error $? "I suspect your system does not have 32-bit development libraries (libc and headers). If you have them, rerun configure with --enable-multilib. If you do not have them, and want to build a 64-bit-only compiler, rerun configure with --disable-multilib." "$LINENO" 5
+#    fi
+#    ;;
+#esac
 
 # Default to --enable-multilib.
 if test x${enable_multilib} = x ; then
