--- gcc-4.9.4/gcc/config/pa/pa.c.bak	2016-01-12 02:41:59.000000000 +0100
+++ gcc-4.9.4/gcc/config/pa/pa.c	2024-01-13 10:06:49.993382141 +0100
@@ -10195,8 +10195,8 @@
     }
 
 #ifdef HAVE_ENABLE_EXECUTE_STACK
-  emit_library_call (gen_rtx_SYMBOL_REF (Pmode, "__enable_execute_stack"),
-		     LCT_NORMAL, VOIDmode, 1, XEXP (m_tramp, 0), Pmode);
+  emit_library_call (gen_rtx_SYMBOL_REF (Pmode, "__enable_execute_stack"),
+		     LCT_NORMAL, VOIDmode, 1, XEXP (m_tramp, 0), Pmode);
 #endif
 }
 
